11_notify/willshen8/pkg/puppy/mapstore.go

```diff
@@ -0,0 +1,65 @@
+package puppy
+
+import (
+	"strconv"
+)
+
+type MapStore struct {
```

Comment:
This implementation of `MapStore` really needs to have a `sync.Mutex`. We haven't worried about it before because there has not been any concurrency in the program, but since lab 10 with the REST interface, the http handlers are called in goroutines so you now have concurrency, and this store can become corrupted without proper locking. It is not hard to add, so I suggest you do. I didn't review your lab 10, but I would have asked for it there. But as this is your last lab, I'd like to see it done correctly.

Reply:
Absolutely agree, and thanks for pointing it out.